packages/telemetry/src/context-aware-slog.js

```diff
@@ -0,0 +1,475 @@
+      case SLOG_TYPES.RUN.START: {
+        if (!(triggerContext || finalBody.runNum === 0)) {
+          const blockTime = finalBody.blockTime || blockContext?.['block.time'];
+
+          assert(blockTime);
+          // TBD: add explicit slog events of both timer poll and install bundle
+          triggerContext = {
+            'run.id': `timer-${finalBody.blockHeight}`,
+            'run.trigger.time': blockTime,
+            'run.trigger.type': 'timer',
+          };
+          persistContext(triggerContext);
+        }
+
+        if (!triggerContext) triggerContext = {};
+        triggerContext = {
+          ...triggerContext,
+          'run.num': `${finalBody.runNum}`,
+        };
+
+        break;
+      }
+      case SLOG_TYPES.SWINGSET.BEGIN_BLOCK: {
+        blockContext = {
+          'block.height': finalBody.blockHeight,
+          'block.time': finalBody.blockTime,
+        };
+        break;
+      }
+      case SLOG_TYPES.SWINGSET.BOOTSTRAP_BLOCK.START: {
+        blockContext = {
+          'block.height': finalBody.blockHeight || 0,
+          'block.time': finalBody.blockTime,
+        };
+        triggerContext = {
+          'run.id': `bootstrap-${finalBody.blockTime}`,
+          'run.trigger.time': finalBody.blockTime,
+          'run.trigger.type': 'bootstrap',
+        };
+        break;
+      }
+      case SLOG_TYPES.SWINGSET.BRIDGE_INBOUND:
+      case SLOG_TYPES.SWINGSET.DELIVER_INBOUND: {
+        const [blockHeight, txHash, msgIdx] = (
+          finalBody.inboundNum || ''
+        ).split('-');
+        const triggerType = body.type.split('-')[2];
+
+        triggerContext = {
+          'run.id': `${triggerType}-${finalBody.inboundNum}`,
+          'run.trigger.blockHeight': Number(blockHeight),
+          'run.trigger.msgIdx': Number(msgIdx),
+          'run.trigger.sender': finalBody.sender,
+          'run.trigger.source': finalBody.source,
+          'run.trigger.time': finalBody.blockTime,
+          'run.trigger.txHash': txHash,
+          'run.trigger.type': triggerType,
+        };
+        persistContext(triggerContext);
+        break;
+      }
+      case SLOG_TYPES.SWINGSET.COMMIT.FINISH:
+      case SLOG_TYPES.SWINGSET.COMMIT.START:
+      case SLOG_TYPES.SWINGSET.END_BLOCK.FINISH: {
+        assert(!!blockContext);
+        break;
+      }
+      case SLOG_TYPES.SWINGSET.END_BLOCK.START: {
+        assert(!!blockContext);
+        break;
+      }
+      case SLOG_TYPES.SYSCALL:
+      case SLOG_TYPES.SYSCALL_RESULT: {
+        logAttributes['crank.syscallNum'] = finalBody.syscallNum;
+
+        delete finalBody.deliveryNum;
+        delete finalBody.replay;
+        delete finalBody.syscallNum;
+
+        break;
+      }
+      default:
+        break;
+    }
+
+    /** @type {LogAttributes} */
+    logAttributes = {
+      ...blockContext,
+      ...crankContext,
+      ...initContext,
+      ...logAttributes,
+      ...replayContext,
+      ...triggerContext,
```

Comment:
Let's order this in order of relevance
```suggestion
      ...logAttributes, // Some static or general ordering info
      ...initContext, // Optional prelude
      ...blockContext, // Block is the first level of execution nesting
      ...triggerContext, // run and trigger info is nested next
      ...crankContext, // Finally cranks are the last level of nesting
      ...replayContext, // Replay is a substitute for crank context during vat page in
```